src/plugin/quarterOfYear/index.js

```diff
@@ -4,7 +4,7 @@ export default (o, c) => {
   const proto = c.prototype
   proto.quarter = function (quarter) {
     if (!this.$utils().u(quarter)) {
-      return this.add((quarter - 1) * 3, M)
+      return this.month(((quarter - 1) * 3) + (this.month() % 3), M)
```

Comment:
Just noticed that you already have a fix here. But the 2nd argument `M` seems unnecessary.